tools/deps.nim

```diff
@@ -28,8 +28,11 @@ proc cloneDependency*(destDirBase: string, url: string, commit = commitHead,
     # from failing
     execRetry fmt"git clone -q {url} {destDir2}"
   if isGitRepo(destDir):
-    execRetry fmt"git -C {destDir2} fetch -q"
-    exec fmt"git -C {destDir2} checkout -q {commit}"
+    let saveDir = getCurrentDir()
+    setCurrentDir(destDir2)
+    execRetry fmt"git fetch -q"
+    exec fmt"git checkout -q {commit}"
+    setCurrentDir(saveDir)
```

Comment:
this won't be called if above code errors; are there other options?
* eg in CI add `apt install git` ?
* if not, then use `defer` at least so the code is correct
* ditto, seems like funs.sh snippet has same issue

won't there be lots of other errors anyways when running on systems that old?

Reply:
> eg in CI add apt install git ?
you mean to update default repositories somehow and the update git to newer version? I'd like to avoid such a complication.

>if not, then use defer at least so the code is correct
ok, done

> ditto, seems like funs.sh snippet has same issue
shell executes commands in order by default, so it cannot omit `co "$OLDPWD"`

Reply:
but `set -e` above would abort on error, wouldn't it? can you try with an intentionally bad commit to double check behavior?

Reply:
sorry, I did not notice `set -e`, you are right.

Reply:
given that the correct code without -C is messy, how about:
```bash
if git version >= ...
  use -C
else:
  fallback to ugly and incorrect code
```

Reply:
it seems it's still safe since change in $PWD occurs only inside parentheses (child shell).

when I run this script from ~/tmp
```bash
myf() {
    (
      set -e
      if test -d ~/bin; then
        cd ~/bin
        echo_run git --unknown
        cd "$OLDPWD"
      fi

      echo never reached
    )
}

echo $PWD
myf
echo $PWD
```

I get original PWD before and after:
```
/home/amakarov/tmp

cmd: git --unknown
unknown option: --unknown
usage: git [--version] [--help] [-C <path>] [-c <name>=<value>]
           [--exec-path[=<path>]] [--html-path] [--man-path] [--info-path]
           [-p | --paginate | -P | --no-pager] [--no-replace-objects] [--bare]
           [--git-dir=<path>] [--work-tree=<path>] [--namespace=<name>]
           <command> [<args>]
/home/amakarov/tmp
```

Reply:
oh, right! that's in fact the reason for the introduction of this subshell, all good then